specification/apimanagement/resource-manager/Microsoft.ApiManagement/preview/2023-03-01-preview/examples/ApiManagementGatewayListTrace.json

```diff
@@ -0,0 +1,112 @@
+{
+  "parameters": {
+    "serviceName": "apimService1",
+    "resourceGroupName": "rg1",
+    "api-version": "2023-03-01-preview",
+    "subscriptionId": "subid",
+    "gatewayId": "gw1",
+    "parameters": {
+      "traceId": "CrDvXXXXXXXXXXXXXVU3ZA2-1"
+    }
+  },
+  "responses": {
+    "200": {
+      "body": {
+        "serviceName": "apimService1",
+        "traceId": "1e0447d4-XXXX-XXXX-XXXX-dbdb8098a0d3",
+        "traceEntries": {
+          "inbound": [
+            {
+              "source": "api-inspector",
+              "timestamp": "2023-05-03T12:03:04.6899436Z",
+              "elapsed": "00:00:00.2983315",
+              "data": {
+                "request": {
+                  "method": "GET",
+                  "url": "https://proxy.msitesting.net/6452XXXXXXXXXXXX9c2facb1/64524dXXXXXXXXXXXX2facb3?subscription-key=117313e70XXXXXXXXXXXX38ba4658ca3",
+                  "headers": [
+                    {
+                      "name": "Host",
+                      "value": "proxy.msitesting.net"
+                    }
+                  ]
+                }
+              }
```

Comment:
Looks like there is potential to return sensitive data form uris, headers, etc.

Is there anything you would be planning to do to allow people to avoid tracing and then accidentally disclosing, sensitive information in their traces?

Reply:
@TimLovellSmith This is a debug functionality. It targets administrators of system to be able to check how request was processed and find if there is any error in that process. To create trace, customer needs to generate debug credentials. The credentials need to be send in the header with request which goes thru APIM gateway proxy. This will allow customer to create trace. Because customer needs to specially craft the request, all possible sensitive data, customer should already know.

The functionality was available in APIM before. Now we want to secure it by applying RBAC security.

@VitaliyKurokhtin Can you please expand on the idea if I am missing something.

Reply:
Yes, this whole API is actually a step towards more secure tracing functionality. At the moment we're solving problem of separating ability to make request, ability to request trace and ability to view resulting trace via separate API actions to allow for granular RBAC control. The second phase of changes will include stripping knowingly secret information from traces - this is not at all trivial given the flexibility we allow for how this information may be manipulated. The third phase would result in inclusion of heuristic tools that would strip away anything that looks like a secret. Since actual content of the trace is left not specified on purpose phases two and three would be non-breaking for customers.